### ISSUE TITLE
3 Suggestion Bars and Progress Bars NOT Working after upgrading from 0.13 to 0.14.1 

### ISSUE THREAD

**Issue by ruancilliers** [learningequality/ka-lite#4603]
Dear Khan Lite Team,

Version: 0.14.1

I was very excited about the new features that was released, specifically the 3 suggestion bars that either directs you to your last exercise and give suggestions. After upgrading to 0.14.1 not everything is working :'( (I installed a new PPA instance as suggestion and migrated from 0.13 as it was installed from Git) 

Expected Bahaviour: I expect all students to get suggestions when you click on the "Ka Lite" Logo and that all students progress bars should be displayed as before in 0.13

Current Behaviour: Very few students get suggestions (mostly when you click on the Ka Lite Logo, nothing happens) and many students the progress reports, under "My Progress" is not working, where it worked in 0.13.

I attached all the different cases as screen shots. Where the progress reports work as well as the suggestions and for students that does not work (which is 90%) of our students. All new students that we create work.

Do you need the database? It is 77MB? I can send it if you want to have a look at it. Let me know what you require...

Ruan.Cilliers@gmail.com
P.S> Your program changed the way we do maths in rural areas. Thank you for your Ka Lite, we depend on it a lot!!! Looking forward to get everything working :-)

![kalite clicked and no recommendations appears asemahle cebani -unexpected error](https://cloud.githubusercontent.com/assets/11680550/10431008/d26b08fa-7102-11e5-8855-d3ee3ee226e1.png)
![kalite clicked and no recommendations appears with unexpected error - uarithmetic_word_problems_1 remember sithungwa](https://cloud.githubusercontent.com/assets/11680550/10431009/d27735d0-7102-11e5-80e4-1d3896729598.png)
![kalite clicked and no recommendations appears asemahle cebani](https://cloud.githubusercontent.com/assets/11680550/10431010/d2856e66-7102-11e5-96f3-39b55a2ef824.png)
![progress bars not working asemahle cebani](https://cloud.githubusercontent.com/assets/11680550/10431011/d28e900e-7102-11e5-81b1-fee7e7d22f8c.png)
![progress bars not working remember sithungwa](https://cloud.githubusercontent.com/assets/11680550/10431012/d32e7c2c-7102-11e5-8783-04789331cca9.png)
![progress bars working hlumisa mavusweni](https://cloud.githubusercontent.com/assets/11680550/10431014/d37c751c-7102-11e5-8f24-9a8f75c81309.png)
![progress bars working for new student mdeni mandible](https://cloud.githubusercontent.com/assets/11680550/10431015/d3864880-7102-11e5-9d70-2d9040768305.png)
![suggestion bars partly working for new student mdeni mandible](https://cloud.githubusercontent.com/assets/11680550/10431016/d3d2fed2-7102-11e5-85c9-04ccb01344d2.png)

**Comment by ruancilliers** [comment]
Forgot to attach the picture where we found one of the few students where the suggestions are working... see below

![everything working hlumisa mavusweni](https://cloud.githubusercontent.com/assets/11680550/10431086/524bb600-7103-11e5-87e7-dce673a56be7.png)

**Comment by rtibbles** [comment]
Hi @ruancilliers - thanks for the detailed reports. Looks like some sort of error is occurring on the backend here that is causing the problem.

If you could upload the database to Dropbox or similar and then share the link with me on richard@learningequality.org, I can take a closer look (and see if this is extant on 0.15 - which we will be releasing shortly).

**Comment by ruancilliers** [comment]
We can also create a teamviewer session if that will be easier for you. As I'm in South Africa the best time will be your early morning so that it can be my late afternoon. Let me know if this is preferred than me sending my 77Mb database for you... Thank you for your help, ruan

**Comment by ruancilliers** [comment]
Fantastic rtibbles! I will upload the database tomorrow (it is now 19h30 in SA) so will be back in the office tomorrow only. Will then send you the link. Thank you for your support :-)

**Comment by ruancilliers** [comment]
Another Feature that I found not working is when you click on the students name's under the coach reports then when it tries to go to the "My Progress" Page of that student it does not work... So the Progress Links are broken from the Coach report...

![coach report links not working](https://cloud.githubusercontent.com/assets/11680550/10449874/b7794856-7196-11e5-8778-4e31241f6268.png)
![coach report links not working - page 2 shows it is not working](https://cloud.githubusercontent.com/assets/11680550/10449875/b77b3e22-7196-11e5-916e-26e4b7eab9e9.png)

**Comment by ruancilliers** [comment]
As requested, please see attached the database link in dropbox. I also included 4 excel files with our students. The grade 6 students are the "new" student created in the beginning of the year. Most of their Suggestion bars are working, where the "old" students seems like it is not working...

All students password is axiumkap (you can just find their user names in the attached excel files)

The super user's username is: axium
Password: P@ssw0rd

https://www.dropbox.com/sh/rmtbebi0bqus8xo/AAAaRN6G9Y6QaPLlUsYW88Xya?dl=0

Hope this helps... let me know if you have any other questions,
Ruan

**Comment by ruancilliers** [comment]
For some reason I get the error message
"Something went really wrong, and we can't process that file. Try again".

Can I mail you the excel files? Please send me a mail so that I can e-mail you the excel files with the user names...
ruan.cilliers@gmail.com
![can t upload excel files](https://cloud.githubusercontent.com/assets/11680550/10450069/0a2aefa4-7198-11e5-8977-bc1f41690227.png)

**Comment by ruancilliers** [comment]
Take note, I also tried to upload PDF's that did not work... So just send me an e-mail and I can send you the excel files with the breakdown of the students that accounts do work and the ones who don't... Thank you!

**Comment by ruancilliers** [comment]
Dear Khan Lite Support,

Any update on the above issue? Did you have time to look at the database yet?

Please let me know if you have any questions.

Thank you,
Ruan

**Comment by MCGallaspy** [comment]
Hi @ruancilliers, I'm now looking at this issue.

**Comment by MCGallaspy** [comment]
This is a legitimate error, and seemingly still exists in version 0.15.x -- not your fault @ruancilliers. If we can come up with a workaround we'll post it here, but in the meantime we'll submit a fix for the next release of KA Lite.

@rtibbles, the error (on branch 0.15.x) with content recommendation is in `content_recommendation.get_exercise_prereqs`. An ExerciseLog has an exercise_id attribute (`arithmetic_word_problems_1`) that no longer exists in the exercise cache. Proposed solution?

Still looking in to the Progress Report stuff.

**Comment by rtibbles** [comment]
Yes, unfortunately, this is due to an issue introduced by insufficient robustness in the content recommendation code.

An interim fix might be to delete the ExerciseLog that has ids that no longer exist.

Will investigate with @MCGallaspy - sorry for the problems here, @ruancilliers.

**Comment by MCGallaspy** [comment]
I like this solution better:

```python
    for exercise_id in exercise_ids:
        exercise = ex_cache.get(exercise_id)
        prereqs += exercise['prerequisites'] if exercise else []
```

And then a pass through the codebase to ensure use of `get_exercise_cache` is similarly robust.

**Comment by rtibbles** [comment]
Hi @ruancilliers - pretty sure we've isolated the issue here. Two different problems in the underlying code, one of which was extant in 0.13, but only came to light because one of the exercises some of your students had engaged with is no longer present in 0.14.1.

We will be issuing the fix to 0.15 and releasing 0.15.1. Upgrading from 0.14.1 to 0.15.1 should work from the PPA. We will flag on here when the fix is released.

Thanks for your patience, and sorry for the delay on our end!

**Comment by rtibbles** [comment]
Fix here should be:

Turn all access of dict attributes in content_recommendation.py to be a ```.get(<attribute>, <default>)```.

Copy changes on lines 91-93 of this commit: https://github.com/learningequality/ka-lite/commit/bfeb2fadd416bbf75e8f2f96d12887a8b3967c0c

**Comment by ruancilliers** [comment]
Dear rtibbles,

Thank you for your help regarding the fix. I really appreciate your help. As recommended I tried to upgrade to version 0.15 from 0.14.2 without success :'( Please see the 6 screen shots attached in how I tried to upgrade to version 0.15. What am I missing? Please advise... As soon as I have managed to upgrade to version 0.15 from 0.14.2 then I will test the fix and let you know...

Thank you!
Ruan

Picture 1&2 - I don't see an option through either the Ubuntu Software Center or the software updates (under "Other Updates", where I normally see any updates for Khan Lite).
Picture 3 - As you can see my current version is 0.14.2
Picture 4 - I execute the sudo apt-get update command successfully with no errors.
Picture 5 - when I execute the command "sudo apt-get install ka-lite", I get the message that the newest package is installed. However when I run the kalite version it is still 0.14.2 (How is this possible?) What am I doing wrong?
Picture 6 - This is information about my environment if you need it.
![picture 1 - neither the ubuntu software system or ppa software updates give option of updating khan lite](https://cloud.githubusercontent.com/assets/11680550/11284604/7929cec8-8f13-11e5-9b56-1b39dd1c3642.JPG)
![picture 2 - neither the ubuntu software system or ppa software updates give option of updating khan lite](https://cloud.githubusercontent.com/assets/11680550/11284605/792e8346-8f13-11e5-9c17-75db173822c9.JPG)
![picture 3 - khan version 14-2](https://cloud.githubusercontent.com/assets/11680550/11284606/79317e70-8f13-11e5-83d8-f194f3453a86.JPG)
![picture 4 - sudo apt-get update successful execution](https://cloud.githubusercontent.com/assets/11680550/11284607/79379c56-8f13-11e5-8ee6-104f6dbe0bc4.JPG)
![picture 5 - sudo apt-get install ka-lite 9already the newest version how is this possible](https://cloud.githubusercontent.com/assets/11680550/11284608/793e3eda-8f13-11e5-95ec-361bb0f0edcb.JPG)
![picture 6 - information about my environment](https://cloud.githubusercontent.com/assets/11680550/11284609/79401246-8f13-11e5-8afe-54ecef2ad96e.JPG)

**Comment by MCGallaspy** [comment]
@ruancilliers the fix has not been released yet. We're planning on releasing 0.15.1 soon, which will incorporate the fix. Upgrading to 0.15.0 (the current latest release) will not resolve the issue.

As for the upgrade issues, did you [subscribe to updates through our PPA](http://ka-lite.readthedocs.org/en/latest/installguide/advanced.html#ppa-installation)? If that does not work, then you can always [download the deb file and follow the manual installation instructions](http://ka-lite.readthedocs.org/en/latest/installguide/install_all.html#linux).

**Comment by rtibbles** [comment]
Hi @ruancilliers - sorry for the delay here. We have now fixed this bug in 0.15.1, upgrading to this version should fix these issues.

Please let us know if it persists, or if you have any other problems!

**Comment by ruancilliers** [comment]
Hi Richard,

Thank you for informing me. I am on leave until 27 December. When I'm back,
I will update the server and let you know.

Enjoy your day! May you have a fantastic festive season :-)
Ruan

On Tuesday, 08 December 2015, Richard Tibbles <notifications@github.com>
wrote:

> Hi @ruancilliers <https://github.com/ruancilliers> - sorry for the delay
> here. We have now fixed this bug in 0.15.1, upgrading to this version
> should fix these issues.
>
> Please let us know if it persists, or if you have any other problems!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/learningequality/ka-lite/issues/4603#issuecomment-163023758>
> .
>


-- 
Sent from Gmail Mobile
